<launch>
  <arg name="climbot_name" default="Climbot"/>
  <arg name="server" default="localhost"/>
  <arg name="serial_port" default="/tmp/vserial_recv"/>

  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node" output="screen">
    <rosparam subst_value="true">
      server: $(arg server)
      port: 3883

      update_frequency: 100.0
      frame_id: world

      # Use the VRPN server's time, or the client's ROS time.
      use_server_time: false
      broadcast_tf: true

      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
      #trackers:
      #- FirstTracker
      #- SecondTracker
    </rosparam>
  </node> 

  <node pkg="NOKOV2ROS" type="pose_vel_pub_node" name="pose_vel_pub_node" args="$(arg climbot_name)" output="screen"/>
  <node pkg="NOKOV2ROS" type="vrpn_node" name="vrpn_node" args="$(arg climbot_name)" output="screen"/>
  <node pkg="All_data_merging" type="serial_parser_pub_node" name="serial_parser_pub_node" args="$(arg serial_port)" output="screen"/>
  <node pkg="All_data_merging" type="cmd_node" name="cmd_node" output="screen"/>
  <node pkg="All_data_merging" type="all_data_pub_node" name="all_data_pub_node" output="screen"/>



</launch>

<!-- 测试方法：先运行python脚本，再运行test.launch文件 -->